ffw/ButtonsRPC.js

```diff
@@ -82,283 +80,179 @@ FFW.Buttons = FFW.RPCObserver.create(
-            'code': 0,
-            'method': 'Buttons.GetCapabilities'
+        case 'Buttons.GetCapabilities': {
+          // send response
+          var JSONMessage = {
+            'jsonrpc': '2.0',
+            'id': request.id,
+            'result': {
+              'capabilities': [...SDL.ButtonCapability, ...SDL.NAVButtonCapability],
+              'presetBankCapabilities': {
+                'onScreenPresetsAvailable': true
+              },
+              'code': SDL.SDLModel.data.resultCode.SUCCESS,
+              'method': 'Buttons.GetCapabilities'
+            }
+          };
+          this.client.send(JSONMessage);
+          break;
+        }
+        case 'Buttons.SubscribeButton': {
+          const { buttonName, appID } = request.params;
+          if(buttonName.includes("NAV_")) {
+            this.navButtonSubscriptionToggle(appID, buttonName, true);
+            const resultCode = SDL.SDLModel.data.resultCode.SUCCESS;
+            console.log("Button " + buttonName + " " + resultCode + " resultCode");
+            this.sendButtonsResult(resultCode, request.id, request.method);
+          } else {
+            const resultCode = this.subscribeButton(appID, buttonName);
+            if(resultCode === 'DO_NOT_RESPOND') {
+              Em.Logger.log('Do not respond on this request');
+              break;
+            }
+            console.log("Button " + buttonName + " " + resultCode + " resultCode");
+            this.sendButtonsResult(resultCode, request.id, request.method);
           }
-        };
-        this.sendMessage(JSONMessage);
+          break;
+        }
+        case 'Buttons.UnsubscribeButton': {
+          const { buttonName, appID } = request.params;
+          if(buttonName.includes("NAV_")) {
+            this.navButtonSubscriptionToggle(appID, buttonName, false);
+            const resultCode = SDL.SDLModel.data.resultCode.SUCCESS;
+            console.log("Button " + buttonName + " " + resultCode + " resultCode");
+            this.sendButtonsResult(resultCode, request.id, request.method);
+          } else {
+            const code = FFW.RPCHelper.getUnsubscribeButtonCustomResultCode(appID, buttonName);
+            if(code === 'DO_NOT_RESPOND') {
+              Em.Logger.log('Do not respond on this request');
+              break;
+            }
+            console.log("Button " + buttonName + " " + code + " Unsubscribe");
+            this.sendButtonsResult(code, request.id, request.method);
+            if (FFW.RPCHelper.isSuccessResultCode(code)) {
+              this.unsubscribeButton(appID, buttonName);
+            }
+          }
+          break;
+        }
       }
     },
+    /**
+     * @function navButtonSubscriptionToggle
+     * @param {Number} appID
+     * @param {String} buttonName
+     * @param {Boolean} subscribe
+     * @description Toggle navigation button subscription by app ID and button name
+     */
+    navButtonSubscriptionToggle(appID, buttonName, subscribe) {
```

Comment:
```suggestion
    setNavButtonSubscription(appID, buttonName, subscribe) {
```
This isn't really a toggle, since you pass the subscription flag in directly

Reply:
@jacobkeeler Renamed 54b806b8f86d11fff6043150b3826d7217489ce3